rclpy/test/test_node.py

```diff
@@ -123,6 +140,40 @@ def test_create_service(self):
         with self.assertRaisesRegex(ValueError, 'unknown substitution'):
             self.node.create_service(GetParameters, 'foo/{bad_sub}', lambda req: None)
 
+    def test_deprecation_warnings(self):
+        warnings.simplefilter('always')
```

Comment:
We discussed offline that it would maybe be nice if `DeprecationWarnings` were turned on by default in rclpy itself, if possible only in rclpy. See the last example in:

https://docs.python.org/3/library/warnings.html#overriding-the-default-filter

Reply:
I couldn't figure out how to get this to work. Including a line similar to that example in the top-level module (or submodule) doesn't seem to work as I expect. Maybe it is sufficient to set a filter to error if there is a warning in our unit tests (e.g. `warnings.simplefilter("error")`).

As for users, we should either [expect them to also be appropriately testing for deprecation warnings](https://docs.python.org/3/library/warnings.html#updating-code-for-new-versions-of-dependencies), or we can use a different type of warning that is not ignored by default.